### PR TITLE
Factory: enhance query_pool

### PR DESCRIPTION
**Pull request by gangov**
Factory: refactors the storage, so it keeps more information about th…e liquidity pool

### CODE REVIEW

File: contracts/factory/src/contract.rs [hunk 1]
```diff
@@ -66,9 +67,46 @@ impl FactoryTrait for Factory {
...
+                address: pool_response.asset_b.address,
+                amount: pool_response.asset_b.amount,
+            },
+            asset_lp_share: Asset {
+                address: pool_response.asset_lp_share.address,
+                amount: pool_response.asset_lp_share.amount,
+            },
+        };
+
+        lp_vec.push_back(lp_to_save);
```
Comment (ueco-jb) [inline contracts/factory/src/contract.rs:94]: Don't save that data in factory.

**Comment by ueco-jb** [comment]
Add two queries:
`query_pool_details(pool_address)`
`query_all_pool_details()` 
Both of those queries invoke `query_pool_info` query on a given pool contract.

**Comment by gangov** [comment]
how are we supposed to test the functionality of the factory deploying multiple liquidity_pools when the lp_wasm_hash is the same every time we deploy a new liquidity_pool?

**Comment by gangov** [comment]
btw what are the fees, something I couldn't understand

**Review by ueco-jb — COMMENTED** [review]
Just some minor adjustments left. The most important would be to attach pool's address to the response.

**Review by ueco-jb — APPROVED** [review]
LGTM :+1:

**Pull request merged**